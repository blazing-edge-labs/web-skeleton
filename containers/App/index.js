import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Router, Link } from '../../routes';
import { logoutAction } from '../../actions/auth';

export class AppComponent extends Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  constructor() {
    super();
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogout(e) {
    const { dispatch } = this.props;

    e.preventDefault();
    dispatch(logoutAction()).then(() => Router.pushRoute('/login'));
  }

  render() {
    const { children } = this.props;

    return (
      <div>
        <nav>
          <Link to="/"><a>Home</a></Link>
          <Link to="/formExample"><a>Components</a></Link>
          <Link to="/profile"><a>Profile</a></Link>
          <a href onClick={this.handleLogout}>Logout</a>
        </nav>
        <div>{children}</div>
      </div>
    );
  }
}

export default connect()(AppComponent);
