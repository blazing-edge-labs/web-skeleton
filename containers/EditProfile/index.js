import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { profileUpdateFetch, changeEmailFetch, changePasswordFetch }
  from 'actions/profile'
import ProfileForm from 'components/ProfileForm'
import ChangeEmailForm from 'components/ChangeEmailForm'
import ChangePasswordForm from 'components/ChangePasswordForm'

export class EditProfileComponent extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };

  constructor() {
    super()
    this.handleProfileUpdate = this.handleProfileUpdate.bind(this)
    this.handleChangeEmail = this.handleChangeEmail.bind(this)
    this.handleChangePassword = this.handleChangePassword.bind(this)
  }

  handleProfileUpdate(values) {
    const { dispatch } = this.props
    return dispatch(profileUpdateFetch(values))
  }

  handleChangeEmail(values) {
    const { dispatch } = this.props
    return dispatch(changeEmailFetch(values))
  }

  handleChangePassword(values) {
    const { dispatch } = this.props

    const newValues = { ...values, confirmation: undefined }
    return dispatch(changePasswordFetch(newValues))
  }

  render() {
    return (
      <main>
        <ProfileForm handleProfileUpdate={this.handleProfileUpdate} />
        <ChangeEmailForm handleChangeEmail={this.handleChangeEmail} />
        <ChangePasswordForm handleChangePassword={this.handleChangePassword} />
      </main>
    )
  }
}

export default connect()(EditProfileComponent)
