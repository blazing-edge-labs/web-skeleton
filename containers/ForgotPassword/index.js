import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Field, reduxForm } from 'redux-form'
import { Link } from 'routes'
import { forgotPasswordFetch } from 'actions/auth'
import { isEmail, isRequired } from 'utils/validator'
import Input from 'components/Input'
import ErrorMsg from 'components/ErrorMsg'
import Button from 'components/Button'

export const validate = (values) => {
  const errors = {}
  const { email } = values

  errors.email = isRequired(email) || isEmail(email)
  return errors
}

export class ForgotPasswordComponent extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    error: PropTypes.string,
    form: PropTypes.string.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    submitSucceeded: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
  };

  constructor() {
    super()
    this.handleSend = this.handleSend.bind(this)
  }

  handleSend(values) {
    const { dispatch } = this.props
    return dispatch(forgotPasswordFetch(values))
  }

  render() {
    const { error, form, handleSubmit, submitSucceeded, submitting } =
      this.props

    return (
      <main>
        <form onSubmit={handleSubmit(this.handleSend)} noValidate>
          <Field
            name="email"
            component={Input}
            id={form}
            label="Email"
            type="email"
            placeholder="Email"
          />
          {submitSucceeded && !submitting && <p>Email has been sent.</p>}
          {error && <ErrorMsg>{error}</ErrorMsg>}
          <Button type="submit" disabled={submitting}>Send</Button>
        </form>
        <Link to="/login"><a>Log in</a></Link>
      </main>
    )
  }
}

export default connect()(reduxForm({
  form: 'ForgotPassword',
  validate,
})(ForgotPasswordComponent))
