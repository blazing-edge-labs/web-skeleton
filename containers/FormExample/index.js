import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';

import Button from '../../components/Button';
import Checkbox from '../../components/Checkbox';
import Input from '../../components/Input';
import Select from '../../components/Select';
import Radio from '../../components/Radio';
import { alwaysError } from '../../utils/validator';
import css from './style.scss';

const options = [
  { label: 'One', value: 1 },
  { label: 'Two', value: 2 },
  { label: 'Three', value: 3 },
];

const theSubmitFunction = (values) => {
  // eslint-disable-next-line
  console.log(values)
};

const FormExample = ({ handleSubmit }) => (
  <div className="formContainer">
    <style jsx>{css}</style>
    <h1>H1 Page Title</h1>
    <p>
      Body Text. Short paragraph intro to the page. Generally 2 or more sentences. Now I am just
      trying to fill more space so that I can have two lines. Yay
    </p>
    <p>
      Body Text. Short paragraph intro to the page. Generally 2 or more sentences. Now I am just
      trying to fill more space so that I can have two lines. Yay
    </p>
    <h2>H2 Page Title</h2>
    <p>
      Body Text. Short paragraph intro to the page. Generally 2 or more sentences. Now I am just
      trying to fill more space so that I can have two lines. Yay
    </p>
    <h1>H1 Page Title</h1>
    <h3>H3 Page Title</h3>
    <p>
      Body Text. Short paragraph intro to the page. Generally 2 or more sentences. Now I am just
      trying to fill more space so that I can have two lines. Yay
    </p>
    <h2>Form Example</h2>
    <form onSubmit={handleSubmit(theSubmitFunction)}>
      <Field
        name="errored"
        label="Try not to turn this into an error"
        validate={[alwaysError]}
        component={Input}
      />
      <Field
        name="shortTextInputDisabled"
        label="Disabled Text Input"
        disabled
        component={Input}
      />
      <Field
        name="shortTextInput2"
        label="Input with Placeholder"
        component={Input}
        placeholder="Here is the placeholder"
      />
      <Field
        name="textarea"
        label="Textarea"
        component={Input}
        textarea
        validate={[alwaysError]}
      />
      <Field
        name="textarea--disabled"
        label="Textarea disabled"
        component={Input}
        textarea
        disabled
      />
      <Field
        name="checkbox1"
        label="The Checkbox"
        component={Checkbox}
      />
      <Field
        name="radio1"
        label="Value 1"
        val="value1"
        component={Radio}
      />
      <Field
        name="radio1"
        label="Value 2"
        val="value2"
        component={Radio}
      />
      <Field
        name="select"
        label="Select"
        component={Select}
        options={options}
        validate={[alwaysError]}
      />
      <Field
        name="select disabled"
        label="Select disabled"
        component={Select}
        options={options}
        disabled
        value={1}
      />
      <Button disabled>Useless</Button>
      <Button>Test</Button>
    </form>
  </div>
);

FormExample.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default connect(() => ({
  initialValues: {
    shortTextInputDisabled: 'Ready only value',
  },
}))(reduxForm({
  form: 'formExample',
})(FormExample));
