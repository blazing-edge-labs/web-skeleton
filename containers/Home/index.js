import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import './style.local.scss';

const HomeComponent = () => {
  return (
    <main>
      <p styleName="test">Home Page!</p>
    </main>
  );
};

// export default HomeComponent;

HomeComponent.propTypes = {
  auth: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
};

export default connect(state => ({
  auth: state.auth,
  user: state.user,
}))(HomeComponent);
