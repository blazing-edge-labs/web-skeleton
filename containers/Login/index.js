import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Field, reduxForm } from 'redux-form'
import { Router, Link } from 'routes'
import { loginFetch } from 'actions/auth'
import { isEmail, isRequired } from 'utils/validator'
import Input from 'components/Input'
import ErrorMsg from 'components/ErrorMsg'
import Button from 'components/Button'

export const validate = (values) => {
  const errors = {}
  const { email, password } = values

  errors.email = isRequired(email) || isEmail(email)
  errors.password = isRequired(password)
  return errors
}

export class LoginComponent extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    error: PropTypes.string,
    handleSubmit: PropTypes.func.isRequired,
    submitting: PropTypes.bool.isRequired,
  };

  constructor() {
    super()
    this.handleLogin = this.handleLogin.bind(this)
  }

  handleLogin(values) {
    const { dispatch } = this.props
    return dispatch(loginFetch(values)).then(() => Router.pushRoute('/'))
  }

  render() {
    const { error, handleSubmit, submitting } = this.props

    return (
      <main>
        <form onSubmit={handleSubmit(this.handleLogin)} noValidate>
          <Field
            name="email"
            component={Input}
            label="Email"
            type="email"
            placeholder="Email"
          />
          <Field
            name="password"
            component={Input}
            label="Password"
            type="password"
            placeholder="Password"
          />
          {error && <ErrorMsg>{error}</ErrorMsg>}
          <Button type="submit" disabled={submitting}>Login</Button>
        </form>
        <Link to="/forgotPassword"><a>Forgot Password?</a></Link>
        <Link to="/signup"><a>Sign up here</a></Link>
      </main>
    )
  }
}

export default connect()(
  reduxForm({
    form: 'Login',
    validate,
  })(LoginComponent),
)
