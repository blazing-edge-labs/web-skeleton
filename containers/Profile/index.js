import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'routes'

export class ProfileComponent extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
  };

  render() {
    const { user } = this.props
    const { email, image } = user

    return (
      <main>
        <dl>
          {image && <dt>Profile Image</dt>}
          {image && (
            <dd>
              <img src={image} alt="Personal Profile" />
            </dd>
          )}
          <dt>Email</dt>
          <dd>{email}</dd>
        </dl>
        <Link to="/editProfile"><a>Edit Profile</a></Link>
      </main>
    )
  }
}

export default connect(state => ({
  user: state.user,
}))(ProfileComponent)
