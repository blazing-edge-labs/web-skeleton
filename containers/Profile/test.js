import React from 'react'
import { shallow } from 'enzyme'
import { ProfileComponent } from './'
import * as Actions from '../../actions/profile'

describe('Profile component', () => {
  const mockDispatch = jest.fn(z => z)
  const wrapper = shallow(
    <ProfileComponent dispatch={mockDispatch} user={{ id: 1 }} />
  )
  const instance = wrapper.instance()
  Actions.profileGetFetch = jest.fn(() => ({}))

  it('componentDidMount method', () => {
    instance.componentDidMount()

    expect(Actions.profileGetFetch).toHaveBeenCalledWith()
    expect(mockDispatch).toHaveBeenCalledWith({})
  })
})
