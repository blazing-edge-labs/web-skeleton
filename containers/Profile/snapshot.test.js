import React from 'react'
import renderer from 'react-test-renderer'
import { ProfileComponent } from 'containers/Profile'

describe('Profile component snapshot', () => {
  it('render all data', () => {
    const tree = renderer.create(
      <ProfileComponent
        dispatch={() => {}}
        user={{
          bio: 'My bio',
          email: 'test@email.com',
          firstname: 'John',
          image: 'http://linkToImage',
          lastname: 'Doe',
        }}
      />
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('render only required email', () => {
    const tree = renderer.create(
      <ProfileComponent
        dispatch={() => {}}
        user={{ email: 'test@email.com' }}
      />
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
