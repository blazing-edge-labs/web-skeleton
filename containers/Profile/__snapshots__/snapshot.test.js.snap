exports[`Profile component snapshot render all data 1`] = `
<main>
  <dl>
    <dt>
      Profile Image
    </dt>
    <dd>
      <img
        alt="Personal Profile"
        src="http://linkToImage" />
    </dd>
    <dt>
      Email
    </dt>
    <dd>
      test@email.com
    </dd>
    <dt>
      First Name
    </dt>
    <dd>
      John
    </dd>
    <dt>
      Last Name
    </dt>
    <dd>
      Doe
    </dd>
    <dt>
      Bio
    </dt>
    <dd>
      My bio
    </dd>
  </dl>
  <a
    onClick={[Function]}
    style={Object {}}>
    Edit Profile
  </a>
</main>
`;

exports[`Profile component snapshot render only required email 1`] = `
<main>
  <dl>
    <dt>
      Email
    </dt>
    <dd>
      test@email.com
    </dd>
  </dl>
  <a
    onClick={[Function]}
    style={Object {}}>
    Edit Profile
  </a>
</main>
`;
